Shape_detection/test/Shape_detection/test_validity_sampled_data.cpp

```diff
@@ -0,0 +1,244 @@
+#include <CGAL/Simple_cartesian.h>
+
+#define CGAL_RANSAC_EXPERIMENTAL_FIXES
+#define USE_WEIGHTED_LEVELS
+#include <CGAL/Shape_detection/Efficient_RANSAC.h>
+#include <CGAL/Shape_detection/Region_growing/Region_growing.h>
+#include <CGAL/Shape_detection/Region_growing/Region_growing_on_point_set.h>
+
+#ifdef CGAL_TEST_RANSAC_PROTOTYPE
+#include <RansacShapeDetector.h>
+#include <CylinderPrimitiveShapeConstructor.h>
+#include <PlanePrimitiveShapeConstructor.h>
+#include <CylinderPrimitiveShape.h>
+#endif
+
+#include <CGAL/IO/read_xyz_points.h>
+#include <CGAL/IO/write_ply_points.h>
+
+#include <CGAL/Real_timer.h>
+
+#include <boost/function_output_iterator.hpp>
+
+namespace SD = CGAL::Shape_detection;
+
+using Kernel = CGAL::Simple_cartesian<double>;
+using Point_3 = Kernel::Point_3;
+using Vector_3 = Kernel::Vector_3;
+
+using Pwn = std::pair<Point_3, Vector_3>;
+using Point_set = std::vector<Pwn>;
+using Point_map = CGAL::First_of_pair_property_map<Pwn>;
+using Normal_map = CGAL::Second_of_pair_property_map<Pwn>;
+
+using RG_query = SD::Point_set::Sphere_neighbor_query<Kernel, Point_set, Point_map>;
+using RG_region = SD::Point_set::Least_squares_plane_fit_region<Kernel, Point_set, Point_map, Normal_map>;
+using Region_growing = SD::Region_growing<Point_set, RG_query, RG_region>;
+
+using RANSAC_traits = SD::Efficient_RANSAC_traits<Kernel, Point_set, Point_map, Normal_map>;
+using RANSAC = SD::Efficient_RANSAC<RANSAC_traits>;
+using RANSAC_plane = SD::Plane<RANSAC_traits>;
+
+void test_copied_point_cloud (const Point_set& points, std::size_t nb);
+
+int main (int argc, char** argv)
+{
+  Point_set points;
+  const char* ifilename = (argc > 1) ? argv[1] : "data/point_set_3.xyz";
+  std::ifstream ifile(ifilename);
+
+  if (!ifile ||
+      !CGAL::read_xyz_points(
+      ifile,
+      std::back_inserter(points),
+      CGAL::parameters::point_map(Point_map()).
+      normal_map(Normal_map())))
+  {
+    std::cerr << "Reading error" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  CGAL::cpp98::random_shuffle (points.begin(), points.end());
+
+  test_copied_point_cloud (points, 1);
+  test_copied_point_cloud (points, 2);
+  test_copied_point_cloud (points, 5);
+  test_copied_point_cloud (points, 10);
+#ifndef CGAL_TEST_SUITE // Disable tests too large for testsuite
+  test_copied_point_cloud (points, 20);
+  test_copied_point_cloud (points, 50);
+#endif
```

Comment:
On my machine, I had to kill the program `./test_validity_sampled_data` because it was consuming 24.5GB of memory. Are you sure that is expected? Maybe there is also a memory leak.

Reply:
Oh, that's weird, I'll look it up.

(Anyway, I think I still need to restrain the test on the testsuite because the tests can be long, even without considering a memory problem.)